api/api.go

```diff
@@ -0,0 +1,121 @@
+package api
+
+import (
+	"errors"
+	"io"
+	"syscall"
+	"time"
+
+	specs "github.com/opencontainers/runtime-spec/specs-go"
+)
+
+var (
+	ErrNotImplemented = errors.New("command not implemented")
+	ErrEmptyID        = errors.New("container id cannot be empty")
+)
+
+type ContainerOperations interface {
+	Create(id string, opts CreateOpts) (*CreateResult, error)
+	Delete(id string, opts DeleteOpts) error
+	Kill(id string, sig syscall.Signal, opts KillOpts) error
```

Comment:
not sure if this is intended to be consumed cross-platform. If yes, probably this can be replaced with `os.Signal` instead of current linux-specific `syscall.Signal`